### ISSUE TITLE
[columns] does not work as expected

### ISSUE THREAD

**Issue by ghost** [uniibu/nbbc#1]
The [columns] tag is not rendered as it should, as in, it doesn't render at all. I checked the implementation of `vanilla/nbbc` and it works exactly as one would expect.

Example text (from the NBBC website):
``` 
This text goes before the columns.
[columns]
This text is in the first column.
This is a second line in the first column.
[nextcol]
This is the second column,
with a second line.
[nextcol]
This is a third column.
Are we having fun yet?
[/columns]
This text goes after the columns.
```

Generated HTML by `vanilla/nbbc`:
``` HTML
<div class="postmsg">This text goes before the columns.
<table class="bbcode_columns"><tbody><tr><td class="bbcode_column bbcode_firstcolumn">
This text is in the first column.<br>
This is a second line in the first column.
</td><td class="bbcode_column">
This is the second column,<br>
with a second line.
</td><td class="bbcode_column">
This is a third column.<br>
Are we having fun yet?
</td></tr></tbody></table>
This text goes after the columns.</div>
```
Generated HTML by `uniibu/nbbc`:
``` HTML
<div class="postmsg">This text goes before the columns.

This text is in the first column.<br>
This is a second line in the first column.
<table class="bbcode_columns"><tbody><tr><td class="bbcode_column">
This is the second column,<br>
with a second line.
</td><td class="bbcode_column">
This is a third column.<br>
Are we having fun yet?
</td></tr></tbody></table>
This text goes after the columns.</div>
```

**Comment by ghost** [comment]
Once this is fixed, could you please create a release, so I don't need to use `dev-master` in my composer.json file? I don't like doing that. :stuck_out_tongue_winking_eye:

**Comment by uniibu** [comment]
Thanks for the report, i will check it out 👍

**Comment by uniibu** [comment]
Hi there, i just checked your issue, this is probably one of the undocumented breaking changes i did. The right code would be.
```
Type your message here.This text goes before the columns.
[columns]
[nextcol]This text is in the first column.
This is a second line in the first column.
[nextcol]This is the second column,
with a second line.
[nextcol]This is a third column.
Are we having fun yet?
[/columns]
This text goes after the columns.
```
Would render
![img](https://i.gyazo.com/f60436544b67d7f30e4768bbcfb293f5.png)

This is to easily separate and differentiate each column

**Comment by ghost** [comment]
Why did you make this change though? It's quite logicial to have the first content after defining `[columns]`. Now it reads like you are creating a table and instantly skipping to the next column. I prefer the old way since it's more logical to me. :stuck_out_tongue:

**Comment by uniibu** [comment]
You are probably right.  On the contrary, if we think of it logically, `[columns]...[/columns]` means we are **defining** elements inside it to be a column and not **declaring** a column it self, so if we set the first element to be the first column without **declaring** it, that would not make sense,  and if thats the case, then the syntax should be `[columns]first column[/columns] [columns]second column[/columns] [columns]third column[/columns]`.

 Now if you argue that `[columns]` is a **declaration** then multiple columns like `[columns]...[/columns][columns]...[/columns]` should look like:
![columnsfirst](https://i.gyazo.com/ee5d7958ff74e896373efd7b224ceb6f.png)
But it does not. it renders like this when defining multiple `[columns]...[/columns]` 
![columns2](https://i.gyazo.com/8d7e66f60e56c546aa1e190812f36b8c.png)

Now having said that, `[columns]...[/columns]` is not a **declaration**  of a column, but rather simply **defining** a block. So i would argue that nbbc's `[columns]first element[nextcol]second element[/columns]` is misleading in a sense that they are misinterpreting a **definition** from **declaration**.  

Tho i would probably change it in the sense that it follows the standard table syntax, since nbbc does not follow the standard `<table><tr><th></th></tr><tr><td></td></tr></table>` . The so called `[columns]..[/columns]` are not really columns, but rather just cells. What nbbc does is basically defining a table's body with a single row, and hacks it way using `<br>` to make it look like a column rather than just a cell. If we visuallize it, it looks like this:
![table1](https://i.gyazo.com/da8b3c52f8a6ec9b1e30e84f29b6b3a9.png)
Which i think is not right. What we want is like this.
![tabl2](https://i.gyazo.com/aed1cb5c25eec9069281d24f3bb94fcb.png)

When i get some time, i would probably document all of this and put a stable release. Thanks for bringing this up. Cheers!

**Comment by ghost** [comment]
You are certainly making valid points! The `columns` syntax is indeed just a table row with some hacky `<br/>`'s to make it work, if you were to change it I would just drop it to be honest as you we already have a way to define proper tables (not sure if they generate proper syntax though). 

The reason I brought this up is because I'm using this parser for a forum which users mostly consist of people that know the "old" NBBC syntax, so they all would have to re-adjust.

**Comment by uniibu** [comment]
If im not mistaken, nbbc does not have a way to define a proper table. There is no [table] syntax in nbbc.

**Comment by uniibu** [comment]
I would probably put columns back to what it was and just add the traditional `[table]` syntax.

**Comment by ghost** [comment]
@uniibu You are totally correct, I must have confused the NBBC BBCode parser with something else.

> I would probably put columns back to what it was and just add the traditional [table] syntax.

I like this idea a lot. :+1: from me!

**Comment by ghost** [comment]
Hey, is there any update on this? :grin:
